subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformOutputs.java

```diff
@@ -35,7 +35,31 @@
 public interface TransformOutputs {
     /**
      * Registers an output directory.
-     * For a relative path, a location for the output directory is provided.
+     *
+     * <p>
+     *     For an <strong>absolute path</strong>, the location is registered as an output directory of the {@link TransformAction}.
+     *     The path is required to point to the {@link InputArtifact}, or point inside the input artifact if it is a directory.
+     *     Example:
+     * </p>
+     * <pre>
+     * Provider&lt;FileSystemLocation&gt; inputArtifact = ...
+     * outputs.dir(inputArtifact.get().getAsFile());
+     * outputs.dir(new File(inputArtifact.get().getAsFile(), "sub-dir"));
+     * </pre>
+     *
+     * <p>
+     *     For a <strong>relative path</strong>, a location for the output directory is provided by Gradle, so that the {@link TransformAction} can produce its outputs at that location.
```

Comment:
Not sure what this means. Does it mean "Gradle creates a temporary output directory into which the transform action must place its output files"?

Reply:
It does mean that, only that the directory is not temporary.